src/build_options.rs

```diff
@@ -152,7 +152,8 @@ impl BuildOptions {
             .filter(|name| name.contains('.'))
             .unwrap_or(&module_name);
 
-        let project_layout = ProjectLayout::determine(manifest_dir, extension_name)?;
+        let py_src = extra_metadata.python_source.clone();
+        let project_layout = ProjectLayout::determine(manifest_dir, extension_name, py_src)?;
```

Comment:
Can avoid the clone here, I think.

```suggestion
        let project_layout = ProjectLayout::determine(
            manifest_dir,
            extension_name,
            extra_metadata.python_source.as_deref()
        )?;
```